### ISSUE TITLE
Adding CORS to downstream SSRS server fails

### ISSUE THREAD

**Issue by nstratton-lfa** [microsoft/reverse-proxy#1803]
### Describe the bug
SQL Server Reporting Services 2016 doesn't offer CORS and a JavaScript fetch fails with a CORS error. I am attempting to add CORS using YARP but failing. Originally posted [1739](https://github.com/microsoft/reverse-proxy/issues/1739) but trying simpler version and opened new issue.

### To Reproduce
program.cs
```
using System.Net;
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddReverseProxy().ConfigureHttpClient((_, handler) =>
{
    handler.Credentials = CredentialCache.DefaultCredentials;
    handler.PreAuthenticate = true;
}).LoadFromConfig(builder.Configuration.GetSection("ReverseProxy"));

var app = builder.Build();
app.UseHttpsRedirection();
app.MapReverseProxy();
app.Run();
```
appSettings.json
```
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  },
  "AllowedHosts": "*",
  "ReverseProxy": {
    "Routes": {
      "route1": {
        "ClusterId": "httpscluster",
        "Match": {
          "Path": "{**catch-all}",
          "Hosts": [ "sec-dev-iis-01.goldcoast.local:8445", "localhost:7027" ]
        },
        "Transforms": [
          {
            "ResponseHeader": "Access-Control-Allow-Origin",
            "Set": "https://my.callingdomain.com"
          }
        ]
      }
    },
    "Clusters": {
      "httpscluster": {
        "HttpRequest": {
          "Version": "1.1",
          "VersionPolicy": "RequestVersionExact"
        },
        "Destinations": {
          "httpsdest": {
            "Address": "https://ssrs.server.org.au/"
          }
        }
      }
    }
  }
}
```
### Further technical details
When I do a JavaScript fetch and look at the network requests.
The transform adds the correct Access-Control-Allow-Origin header to the first request headers from YARP and it is allowed. 
The response headers for the second request are directly from the ssrs server, not YARP and fail with a 401.
www-authenticate: Negotiate
www-authenticate: Kerberos
www-authenticate: NTLM

If I use YARP to browse to the ssrs server, the credentials of the AccountPool are shown as the user connecting to ssrs and all is well but the fetch is failing with a CORS error.

Access to fetch at 'https://ssrs.server.org.au/ReportServer?/FlightOpsReports/InwardRequest&rs:Command=Render&rs:ClearSession=true&rs:Format=PDF&ID=2019' (redirected from 'https://sec-dev-iis-01.goldcoast.local:8445/reportserver/Pages/ReportViewer.aspx?/Reports/InwardRequest&rs:Command=Render&rs:ClearSession=true&rs:Format=PDF&ID=2019') from origin 'https://my.callingdomain.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

- Windows Server 2019
- IIS 10
- YARP 1.1.0
- Microsoft.AspNetCore.Authentication.Negotiate 6.0.3

**Comment by adityamandaleeka** [comment]
Can you share a Fiddler trace? Perhaps the backend server is somehow redirecting the client in a way that bypasses the proxy, causing this type of issue. Having the trace might help us understand what's happening.

**Comment by nstratton-lfa** [comment]
Hi,
Thanks so much for your response. In light of a tight time schedule, I'll close this because I have re-written it with a minimal api that serves the requests. It's a more restrictive solution but it works for now.
Kind Regards
